scripts/autotools/libmongocrypt/CheckSSL.m4

```diff
@@ -0,0 +1,35 @@
+if test "$PHP_CLIENT_SIDE_ENCRYPTION" != "no"; then
+  AC_MSG_CHECKING([which crypto library to use for libmongocrypt])
+  AC_MSG_RESULT([$PHP_MONGODB_SSL])
+
+  dnl Disable Windows crypto
+  AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_CNG, 0)
+
+  if test "$PHP_MONGODB_SSL" = "darwin"; then
+    PHP_CLIENT_SIDE_ENCRYPTION="yes"
+
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_LIBCRYPTO, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
+
+    PHP_MONGODB_LIBMONGOCRYPT_CFLAGS="$PHP_MONGODB_LIBMONGOCRYPT_CFLAGS -DKMS_MESSAGE_ENABLE_CRYPTO=1 -DKMS_MESSAGE_ENABLE_CRYPTO_COMMON_CRYPTO=1"
```

Comment:
These lines look new, but perhaps I just didn't notice them in previous iterations of the PR. I suppose these custom defines were added to balance the logic in `kms-message/CMakeLists.txt`.

Autotools does provide an [`AC_DEFINE`](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Defining-Symbols.html) macro. Can we use that instead? Note that the value defaults to `1` if not specified, but the docs suggest not depending on that behavior, so we could do:

```
AC_DEFINE([KMS_MESSAGE_ENABLE_CRYPTO], 1)
AC_DEFINE([KMS_MESSAGE_ENABLE_CRYPTO_COMMON_CRYPTO], 1)
```

...and similar for the case below.

Reply:
I had previously tried this but it didn't work, but revisiting this the following `AC_DEFINE` worked:
```
    AC_DEFINE([KMS_MESSAGE_ENABLE_CRYPTO], 1, [ ])
    AC_DEFINE([KMS_MESSAGE_ENABLE_CRYPTO_COMMON_CRYPTO], 1, [ ])
```

Reply:
On second try, this still doesn't work: while this creates `#define` instructions in `config.h`, testing FLE causes exceptions due to missing crypto:
> Operation threw unexpected MongoDB\Driver\Exception\BulkWriteException: Bulk write failed due to previous MongoDB\Driver\Exception\RuntimeException: failed to create KMS message

Reverting to `-DKMS_MESSAGE_ENABLE_CRYPTO` fixes this.

Reply:
Ah, I forgot that `AC_DEFINE` sets stuff through the generated `config.h` file. Looking at `ax_pthread.m4` jogged my memory. In that case, CFLAGS is the way to go.